tests/shutdown/cleanup_before_shutdown.pm

```diff
@@ -67,8 +67,12 @@ END_SCRIPT
             systemctl 'stop wickedd.service';
             assert_script_run('ls /var/lib/wicked/');
             save_screenshot;
-            assert_script_run('rm -f /var/lib/wicked/*.xml');
+            script_run('rm -f /var/lib/wicked/*.xml');
+            script_run('rm -f /var/run/wicked/extension/hostname');
+            script_run('rm -f /var/run/wicked/extension/hostname');
+            script_run('rm -f /var/run/wicked/*');
         }
+        script_run('rm -f /etc/hostname');
```

Comment:
Does this apply to Tumbleweed as well?

Reply:
Maybe it is better to tell me the story behind it again, sorry for asking twice.

Reply:
Since installation scenarios mainly creating images for use by other jobs , and in some cases same qcow used for multiply tests sometimes when they are running on same VLAN ( MM tests ) it is actually better to delete this file . It is supported use case when there is no hostname file system will not crash but should just pick up random name

Reply:
but I am not sure about this line , I think we need to add some condition here to not execute it everywhere . this is why PR has WIP . want to ask others what they think before set something . If feedback would be too negative I would end up with `if(check_var('EXTRATEST','wicked'))` :)

Reply:
@okurz  , @DimStar77  what do you think about this ? is it bad idea ?

Reply:
> but I am not sure about this line , I think we need to add some condition here to not execute it everywhere . this is why PR has WIP . want to ask others what they think before set something . If feedback would be too negative I would end up with `if(check_var('EXTRATEST','wicked'))` :)

Looks reasonable for me, but also wonder what @okurz @DimStar77 @rwx788 think about it.

Reply:
By now we know: it's a bad idea: https://openqa.opensuse.org/tests/1150003#step/gcc_compilation/13

Reply:
@asmorodskyi Thinking about it twice, it looks dangerous to me to delete `/etc/hostname`, which is expected by many software. BTW you didn't explain in commit message why you do it (so we can just guess the reason).

Reply:
yeah there is already reverting PR #9389 with ongoing discussion ...